core/src/hir/types.rs

```diff
@@ -20,6 +20,7 @@ pub enum Type<P: TyPosition = Everywhere> {
     Primitive(PrimitiveType),
     Opaque(OpaquePath<Optional, P::OpaqueOwnership>),
     Struct(P::StructPath),
+    Trait(TraitPath),
```

Comment:
nit: traits are only allowed as input types, so this should be P::TraitPath where it resolves to an empty enum or TraitPath